### DISCUSSION TITLE
Generate Email Template

### DISCUSSION THREAD

**Discussion by jsalgado10** [OrchardCMS/OrchardCore#8340] [Q&A] [answered]
I am working on a notifications module that will send an email to users on different types of events. I want to setup a template for each action and pass the data required to fill in the information. Then get the result html and add it to the MailMessage object.

This is a solution I found but it seems that is for OchardCMS only

https://dev.to/niranjankala/how-to-send-email-in-orchard-cms-7nf

Here is the code that it says to use

`` var template = _shapeFactory.Create("Template_User_Validated", Arguments.From(new {
    RegisteredWebsite = site.As<RegistrationSettingsPart>().ValidateEmailRegisteredWebsite,
    ContactEmail = site.As<RegistrationSettingsPart>().ValidateEmailContactEMail,
    ChallengeUrl = url
}));``
`` template.Metadata.Wrappers.Add("Template_User_Wrapper");  ``

Is this possible on Orchard Core?

**Comment by Skrypt** [comment]
You can use Liquid Templates now so the code might be different. Though the principle should work.

**Reply by hishamco** [reply]
You can use Razor too ;)

**Reply by jsalgado10** [reply]
Sorry for the confusion. I am wondering which class and method I can use to pass in  a liquid/razor template and their parameters and return the html result.

**Comment by agriffard** [comment, marked as answer]
https://github.com/OrchardCMS/OrchardCore/blob/c54b20a437ce8498deaa65e5a0af14c014f3169b/src/OrchardCore.Modules/OrchardCore.Users/Controllers/ControllerExtensions.cs#L24

**Reply by jsalgado10** [reply]
@agriffard this is exactly what i was looking for.

Thanks
